rocket-cookie-auth/src/user.rs

```diff
@@ -0,0 +1,76 @@
+use crate::auth::rand_string;
+use crate::error::Error;
+use anyhow::Result;
+use argon2::verify_encoded;
+use rocket::http::Status;
+use rocket::request::FromRequest;
+use rocket::request::Outcome;
+use rocket::Request;
+use serde::Deserialize;
+use serde::Serialize;
+
+/// A request guard that can be included in handler definitions to enforce authentication.
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone, Hash, PartialOrd, Ord)]
+pub struct User {
+    pub id: u32,
+    #[serde(skip_serializing)]
+    pub(crate) password: String,
```

Comment:
Do we want to include a hash scheme version in case of some future issue so that old passwords can be noted to be updated to the new storage scheme?

Reply:
Interesting idea. How would that look like?

Reply:
It is actually, aragon the library is taking care of this. The `password` might look like this: `$argon2i$v=19$m=65536,t=3,p=1$YOtX2//7NoD/owm8RZ8llw==$fPn4sPgkFAuBJo3M3UzcGss3dJysxLJdPdvojRF20ZE=` which will include all the bits and pieces needed. 

Checkout this [stackoverflow answer](https://security.stackexchange.com/questions/222744/which-part-of-this-encoded-argon2-hash-is-the-salt)

Reply:
Neat! I assume in future if we want to change from argon2 to another hash (for whatever future reason), that encoded form could also include some marker of how it was encoded.